plugins/temporal/rpc.go

```diff
@@ -0,0 +1,492 @@
+ListOpenWorkflow(ctx context.Context, request *workflowservice.ListOpenWorkflowExecutionsRequest) (*workflowservice.ListOpenWorkflowExecutionsResponse, error)
+ListWorkflow(ctx context.Context, request *workflowservice.ListWorkflowExecutionsRequest) (*workflowservice.ListWorkflowExecutionsResponse, error)
+ListArchivedWorkflow(ctx context.Context, request *workflowservice.ListArchivedWorkflowExecutionsRequest) (*workflowservice.ListArchivedWorkflowExecutionsResponse, error)
+ScanWorkflow(ctx context.Context, request *workflowservice.ScanWorkflowExecutionsRequest) (*workflowservice.ScanWorkflowExecutionsResponse, error)
+CountWorkflow(ctx context.Context, request *workflowservice.CountWorkflowExecutionsRequest) (*workflowservice.CountWorkflowExecutionsResponse, error)
+etSearchAttributes(ctx context.Context) (*workflowservice.GetSearchAttributesResponse, error)
+QueryWorkflowWithOptions(ctx context.Context, request *QueryWorkflowWithOptionsRequest) (*QueryWorkflowWithOptionsResponse, error)
+DescribeTaskQueue(ctx context.Context, taskqueue string, taskqueueType enumspb.TaskQueueType) (*workflowservice.DescribeTaskQueueResponse, error)
+*/
+
+const name = "rpc"
+
+type EmptyStruct struct{}
+
+type StartWorkflowOptions struct {
+	// ID - The business identifier of the workflow execution.
+	// Optional: defaulted to a uuid.
+	ID string `json:"wid,omitempty"`
+
+	// TaskQueue - The workflow tasks of the workflow are scheduled on the queue with this name.
+	// This is also the name of the activity task queue on which activities are scheduled.
+	// The workflow author can choose to override this using activity options.
+	// Mandatory: No default.
+	TaskQueue string `json:"task_queue"`
+
+	// WorkflowExecutionTimeout - The timeout for duration of workflow execution.
+	// It includes retries and continue as new. Use WorkflowRunTimeout to limit execution time
+	// of a single workflow run.
+	// The resolution is seconds.
+	// Optional: defaulted to 10 years.
+	WorkflowExecutionTimeout time.Duration `json:"workflow_execution_timeout"`
+
+	// WorkflowRunTimeout - The timeout for duration of a single workflow run.
+	// The resolution is seconds.
+	// Optional: defaulted to WorkflowExecutionTimeout.
+	WorkflowRunTimeout time.Duration `json:"workflow_run_timeout"`
+
+	// WorkflowTaskTimeout - The timeout for processing workflow task from the time the worker
+	// pulled this task. If a workflow task is lost, it is retried after this timeout.
+	// The resolution is seconds.
+	// Optional: defaulted to 10 secs.
+	WorkflowTaskTimeout time.Duration `json:"workflow_task_timeout"`
+}
+
+/*
+- the method's type is exported.
+- the method is exported.
+- the method has two arguments, both exported (or builtin) types.
+- the method's second argument is a pointer.
+- the method has return type error.
+*/
+type Rpc struct {
+	client client.Client
+}
+
+func (p *Provider) Name() string {
```

Comment:
does it make sense to move it into temporal.go file?

Reply:
sure, it does. done